src/AaveCurator/AaveCurator.sol

```diff
@@ -0,0 +1,310 @@
+  }
+
+  function removeManager() external onlyAdmin {
+    address oldManager = manager;
+    manager = address(0);
+    emit ManagerChanged(oldManager, manager);
+  }
+
+  function setAllowedFromToken(address token, address oracle, bool isEthBased, bool allowed) external onlyAdmin {
+    if (token == address(0)) revert Invalid0xAddress();
+    if (oracle == address(0)) revert Invalid0xAddress();
+    allowedFromTokens[token] = allowed;
+    tokenChainlinkOracle[token] = TokenChailinkOracleData(oracle, isEthBased);
+    emit TokenUpdated(token, allowed);
+  }
+
+  function setAllowedToToken(address token, address oracle, bool isEthBased, bool allowed) external onlyAdmin {
+    if (token == address(0)) revert Invalid0xAddress();
+    if (oracle == address(0)) revert Invalid0xAddress();
+    allowedToTokens[token] = allowed;
+    tokenChainlinkOracle[token] = TokenChailinkOracleData(oracle, isEthBased);
+    emit TokenUpdated(token, allowed);
+  }
+
+  function setMilkmanAddress(address _milkman) external onlyAdmin {
+    if (_milkman == address(0)) revert Invalid0xAddress();
+    milkman = _milkman;
+  }
+
+  /// @notice Transfer any tokens on this contract to Aave V3 Collector
+  /// @param tokens List of token addresses
+  function withdrawToCollector(address[] calldata tokens) external onlyAdminOrManager {
+    for (uint256 i = 0; i < tokens.length; ++i) {
+      IERC20(tokens[i]).safeTransfer(
+        address(AaveV3Ethereum.COLLECTOR),
+        IERC20(tokens[i]).balanceOf(address(this))
+      );
+    }
+  }
+
+  receive() external payable {}
+
+  /// @inheritdoc VersionedInitializable
+  function getRevision() internal pure override returns (uint256) {
+    return REVISION;
+  }
+
+  function _getPriceCheckerAndData(
+    TokenType _tokenType,
+    uint256 slippage,
+    address fromToken,
+    address toToken
+  ) internal returns (address, bytes memory) {
+    if (_tokenType == TokenType.Standard) {
+      return (CHAINLINK_PRICE_CHECKER, _getChainlinkCheckerData(slippage, fromToken, toToken));
+    } else if (_tokenType == TokenType.BPT8020) {
+      return (BPT_PRICE_CHECKER, _getBPTCheckerData(slippage, fromToken, toToken));
+    } else {
+      revert InvalidOracleType();
+    }
+  }
+
+  function _getChainlinkCheckerData(
+    uint256 slippage,
+    address fromToken,
+    address toToken
+  ) internal view returns (bytes memory) {
+    TokenChailinkOracleData memory oracleOne = tokenChainlinkOracle[fromToken];
+    TokenChailinkOracleData memory oracleTwo = tokenChainlinkOracle[toToken];
+
+    if (oracleOne.oracle == address(0) || oracleTwo.oracle == address(0)) revert OracleNotSet();
+
+    bytes memory data;
+    if (oracleOne.isEthBased != oracleTwo.isEthBased) {
+        address[] memory paths = new address[](3);
+        paths[0] = oracleOne.oracle;
+        paths[1] = AaveV3EthereumAssets.WETH_ORACLE;
+        paths[2] = oracleTwo.oracle;
+
+        bool[] memory reverses = new bool[](3);
```

Comment:
I *think* (but don't take my word on it) that you need to set reverses[1] = true iff the from token has a USD oracle